### DISCUSSION TITLE
Handling Mixed Quality at Providers

### DISCUSSION THREAD

**Discussion by AJarOfDirt0** [Opinionated-Blueprints/10-Books-Full-of-Rails#31] [Ideas]
I decided to try throwing some quality modules in my miners on fulgora and this immediately locked up the train since it didn't have a full stack of one of the higher quality ores.

**Comment by AJarOfDirt0** [comment]
I added a simple inactivity check to the interrupt, but maybe there is a better solution?

**Comment by Bocian-1** [comment]
Not yet, I haven't looked into transporting mixed items yet

**Comment by Bocian-1** [comment]
Added support for mixed items as of v2.17.0

**Comment by AlexCPU** [comment]
I'm currently on v2.17.3 and had a bit of fun trying to move around scrap of 3 different qualities from the same provider to the same requestors.  The providers locked up as they weren't able to fill the train (non-full stacks of rare scrap), so I added an idle timeout to the trains interrupts for providers.  On the requestor stations I had to add the uncommon and rare qualities to the constant combinator to get the other qualities to unload.  I still ran in to a problem where a train was dispatched to a station that didn't _quite_ have enough space to store it due to the mix of qualities and stack sizes.  I haven't solved that yet beyond just starting consuming the scrap so that the station drains whilst the train is waiting.

So from what I've seen, I think a mixed train will get unloaded at multiple stations (one item per station), but I'm not sure how to load mixed items in the first place, or unload them at a single station "properly".

**Reply by Bocian-1** [reply]
Looks like I forgot to update the train blueprints with interrupts modified for mixed cargo. v2.18.1 has the correct interrupts now. It's probably the same or similar to what you did.

For providers, it should be enough to just send in mixed belts.
For requesters, the way I implemented it you can't unload multiple items at a single requester properly.
That would require a separate system, which I may implement later. The main advantage of the current one is the stations don't need any special treatment from you. No need for a separate group or any values in combinators or separate blueprints. Just put in mixed resources and it should work.

I have a working setup loading stack inserter of every quality and distributing it across multiple stations (the same as in the preview videos) on my testing save file. You should be able to load it without any of the mods, but I'd recommend keeping editor extensions and creative mode
[SA RAIL.zip](https://github.com/user-attachments/files/17821560/SA.RAIL.zip)
